src/main/java/hudson/plugins/gearman/MyGearmanWorkerImpl.java

```diff
@@ -634,7 +634,8 @@ private void submitFunction(GearmanFunction fun) {
             if (executorService == null) {
                 fun.call();
             } else {
-                executorService.submit(fun);
+                // The future is not used, we watch it via driveSessionIO()
+                Future task = executorService.submit(fun);
```

Comment:
Instead of having a dead store to make spotbugs happy, it might be better to use @SuppressFBWarnings.

Even better, would be to use Executor.execute(), but it would require wrapping that Callable in a Runnable.

Reply:
It is a bit more complicated, the executorService is apparently never used. `MyGearmanWorkerImpl` is always called with a single argument and reinvoke itself using null as an executorService:
```  
      public MyGearmanWorkerImpl(AvailabilityMonitor availability) {
          this (null, availability);
      }
```
If I replace the `submit()` with an exception, the tests still pass which tend to confirm an ExecutorService is never used! Line 360 above there is a comment stating the author planned to run the job asynchronously in the future:
```
              // For the time being we will execute the jobs synchronously
              // in the future, I expect to change this.
              if (function != null) {
                  LOG.debug("---- Worker " + this + " executing function");
                  submitFunction(function);
```

So essentially the code always has a `null` executor and thus always invoke `fun.call()` directly and never use the ExecutorService. I traced the comment back to the change that introduced the class https://review.opendev.org/c/x/gearman-plugin/+/26911  I guess they had a plan to use ExecutorService and wrote a bunch of boilerplate in advance :)